pytorch_lightning/loops/utilities.py

```diff
@@ -0,0 +1,105 @@
+# Copyright The PyTorch Lightning team.
```

Comment:
@awaelchli @carmocca do you recommend any indication for marking that these apis are utilities for the trainer/batch loop? users shouldn't call these directly

Reply:
> users shouldn't call these directly

So we don't want loop users to call these directly
But we do allow loop writers to do so.

I don't know how we could indicate this other than via docstrings.

Reply:
something like `loops/_utilities.py` maybe?

Reply:
> why should they not call it?

It's beneficial to draw a clear distinction between public APIs and internal implementation details. We can evolve the latter without worrying about breaking user code.

Reply:
Yes, I spend my time accessing private utils in PyTorch and wished this wasn't the case. 
I believe even internal implementation details should be designed in the way that advanced users will access it.